### PR TITLE
FIX: COPY-AWS: using datastore key

### PR DESCRIPTION
**Pull request by nicolas2bert**
Do not need to re-create the source key name if we get it directly from metadata.location info :)

### CODE REVIEW

**Comment by nicolas2bert** [comment]
@ironman-machine try

File: lib/data/external/AwsClient.js [hunk 1]
```diff
@@ -359,26 +359,21 @@ class AwsClient {
             return callback();
         });
     }
-    copyObject(request, sourceBucketName, sourceObjectKey, sourceLocation,
-    log, callback) {
+    copyObject(request, sourceKey, sourceLocation, log, callback) {
```
Comment (LaurenSpiegel) [inline lib/data/external/AwsClient.js]: what's the difference between sourceKey and sourceLocation?

File: lib/data/external/AwsClient.js [hunk 2]
```diff
@@ -359,26 +359,21 @@ class AwsClient {
...
         const destBucketName = request.bucketName;
         const destObjectKey = request.objectKey;
         const destAwsKey = this._createAwsKey(destBucketName, destObjectKey,
           this._bucketMatch);
 
-        const sourceBucketMatch =
-        config.locationConstraints[sourceLocation].details.bucketMatch;
```
Comment (LaurenSpiegel) [inline lib/data/external/AwsClient.js]: do you not need this because the key already has the info?

File: lib/data/multipleBackendGateway.js [hunk 3]
```diff
@@ -228,12 +228,11 @@ const multipleBackendGateway = {
...
     // backend to the same external backend
-    copyObject: (request, location, sourceBucket, sourceObject,
-    sourceLocation, log, cb) => {
+    copyObject: (request, location, sourceKey, sourceLocation, log, cb) => {
         const client = clients[location];
         if (client.copyObject) {
-            return client.copyObject(request, sourceBucket, sourceObject,
-            sourceLocation, log, (err, key) => {
+            return client.copyObject(request, sourceKey, sourceLocation, log,
+            (err, key) => {
```
Comment (LaurenSpiegel) [inline lib/data/multipleBackendGateway.js]: add a test please

File: lib/data/external/AwsClient.js (hunk 2 above)
Comment (nicolas2bert) [reply lib/data/external/AwsClient.js]: yes :)

File: lib/data/external/AwsClient.js (hunk 1 above)
Comment (nicolas2bert) [reply lib/data/external/AwsClient.js]: `sourceKey` is the AWS source object name, Maybe I should call it `awsSourceKey`
`sourceLocation` is the location of the source object

Comment (LaurenSpiegel) [reply lib/data/external/AwsClient.js]: I still am not clear on the difference.

Comment (nicolas2bert) [reply lib/data/external/AwsClient.js]: In this case: `awsSourceKey` === `bucketname/objectname` and `sourceLocation` === `aws-test`

Comment (nicolas2bert) [reply lib/data/external/AwsClient.js]: locationConstraintName

**Pull request merged**